Desenvolver o arquivo de Teste

Ao executar o action inicial dos testes o seguinte erro apareceu

```
============================ no tests ran in 0.00s =============================
Error: Process completed with exit code 5.
```

Significa que o **pytest** foi executado, mas **nenhum teste foi encontrado** no projeto. O código de saída `exit code 5` normalmente indica que o `pytest` não encontrou nenhum teste para rodar.

### Possíveis causas e soluções para esse erro:

1. **Nenhum arquivo de teste foi encontrado**:
   O `pytest` por padrão procura por arquivos que começam com `test_` ou terminam com `_test.py`. Obviamente ainda não fizemos isso, mas precisamos discutir como proceder aqui nessa issue.

   **Sugestão**: 
   - `tests_CI.py`

2. **Localização dos testes**:
   O `pytest` pode não estar encontrando os testes devido ao diretório onde ele está sendo executado. Por padrão, ele começa a busca a partir do diretório onde é executado (geralmente a raiz do repositório). Se formos usar os testes  em uma pasta específica (por exemplo, `tests/`), isso deve ser especificado.

   **Solução**: Especificação explicita da pasta onde os testes estão localizados:

   ```yaml
   - name: Run tests
     run: |
       pytest tests/  # Substitua 'tests/' pelo diretório correto
   ```

3. **Formato dos testes**:
   Certifiquem-se de usar funções de teste que o `pytest` reconhece. Isso vai dar um pouco de trabalho e exigir alguns estudos da gente. As funções de teste devem começar com `test_` para serem detectadas automaticamente. Por exemplo:

   ```python
   # Exemplo de um arquivo de teste
   def test_example():
       assert 1 + 1 == 2
   ```

   Se os testes não seguirem esse padrão, o `pytest` pode não identificá-los como testes.

### Como diagnosticar e corrigir:

1. **Verifique se seus testes têm o nome correto** (começando com `test_` ou terminando com `_test.py`).
2. **Especifique explicitamente o diretório de testes** se eles estiverem em um diretório específico, como `tests/`.
3. **Verifique se as funções de teste estão corretamente definidas** e começam com `test_`.

--- a/tests/test_und_conv.py
+++ b/tests/test_und_conv.py
@@ -1,4 +1,5 @@
 import pytest
+from pytest import approx
 from und_fisicas import *
 
 ####################### 1. Testes de Temperatura #######################
@@ -33,17 +34,17 @@ def test_kelvin_celsius():
 
 def test_fahrenheit_kelvin():
     """Testa a conversão de Fahrenheit para Kelvin"""
-    assert fahrenheit_kelvin(32) == 273.15  # Ponto de congelamento da água
-    assert fahrenheit_kelvin(212) == 373.15  # Ponto de ebulição da água
-    assert fahrenheit_kelvin(-40) == 233.15  # Teste com valores negativos
-    assert fahrenheit_kelvin(98.6) == 310.15  # Temperatura corporal média
+    assert fahrenheit_kelvin(32) == approx(273.15, rel=1e-4)  # Ponto de congelamento da água
+    assert fahrenheit_kelvin(212) == approx(373.15, rel=1e-4)  # Ponto de ebulição da água
+    assert fahrenheit_kelvin(-40) == approx(233.15, rel=1e-4)  # Teste com valores negativos
+    assert fahrenheit_kelvin(98.6) == approx(310.15, rel=1e-4)  # Temperatura corporal média
 
 def test_kelvin_fahrenheit():
     """Testa a conversão de Kelvin para Fahrenheit"""
-    assert kelvin_fahrenheit(273.15) == 32  # Ponto de congelamento da água
-    assert kelvin_fahrenheit(373.15) == 212  # Ponto de ebulição da água
-    assert kelvin_fahrenheit(0) == -459.67  # Zero absoluto
-    assert kelvin_fahrenheit(310.15) == 98.6  # Temperatura corporal média
+    assert kelvin_fahrenheit(273.15) == approx(32, rel=1e-4)  # Ponto de congelamento da água
+    assert kelvin_fahrenheit(373.15) == approx(212, rel=1e-4)  # Ponto de ebulição da água
+    assert kelvin_fahrenheit(0) == approx(-459.67, rel=1e-4)  # Zero absoluto
+    assert kelvin_fahrenheit(310.15) == approx(98.6, rel=1e-4)  # Temperatura corporal média
 
 ####################### 2. Testes de Distância #######################
 
@@ -63,15 +64,15 @@ def test_quilometros_metros():
 
 def test_milhas_quilometros():
     """Testa a conversão de milhas para quilômetros"""
-    assert milhas_quilometros(1) == 1.60934  # 1 milha = 1.60934 quilômetros
-    assert milhas_quilometros(5) == 8.0467  # 5 milhas = 8.0467 quilômetros
+    assert milhas_quilometros(1) == approx(1.60934, rel=1e-4)  # 1 milha = 1.60934 quilômetros
+    assert milhas_quilometros(5) == approx(8.0467, rel=1e-4)  # 5 milhas = 8.0467 quilômetros
     assert milhas_quilometros(0) == 0  # 0 milhas = 0 quilômetros
-    assert milhas_quilometros(0.5) == 0.80467  # Valor não inteiro
+    assert milhas_quilometros(0.5) == approx(0.80467, rel=1e-4)  # Valor não inteiro
 
 def test_quilometros_milhas():
     """Testa a conversão de quilômetros para milhas"""
-    assert quilometros_milhas(1) == 0.621371  # 1 quilômetro = 0.621371 milhas
-    assert quilometros_milhas(5) == 3.106855  # 5 quilômetros = 3.106855 milhas
+    assert quilometros_milhas(1) == approx(0.621371, rel=1e-4)  # 1 quilômetro = 0.621371 milhas
+    assert quilometros_milhas(5) == approx(3.106855, rel=1e-4)  # 5 quilômetros = 3.106855 milhas
     assert quilometros_milhas(0) == 0  # 0 quilômetros = 0 milhas
     assert quilometros_milhas(1.60934) == 1  # 1 milha = 1 quilômetro
 
@@ -79,24 +80,24 @@ def test_quilometros_milhas():
 
 def test_quilos_libras():
     """Testa a conversão de quilogramas para libras"""
-    assert quilos_libras(1) == 2.20462  # 1 quilograma = 2.20462 libras
-    assert quilos_libras(5) == 11.0231  # 5 quilogramas = 11.0231 libras
+    assert quilos_libras(1) == approx(2.20462, rel=1e-4)  # 1 quilograma = 2.20462 libras
+    assert quilos_libras(5) == approx(11.0231, rel=1e-4)  # 5 quilogramas = 11.0231 libras
     assert quilos_libras(0) == 0  # 0 quilograma = 0 libras
-    assert quilos_libras(10.5) == 23.1495  # Valor não inteiro
+    assert quilos_libras(10.5) == approx(23.1495, rel=1e-4)  # Valor não inteiro
 
 def test_libras_quilos():
     """Testa a conversão de libras para quilogramas"""
-    assert libras_quilos(2.20462) == 1  # 2.20462 libras = 1 quilograma
-    assert libras_quilos(11.0231) == 5  # 11.0231 libras = 5 quilogramas
+    assert libras_quilos(2.20462) == approx(1, rel=1e-4)  # 2.20462 libras = 1 quilograma
+    assert libras_quilos(11.0231) == approx(5, rel=1e-4)  # 11.0231 libras = 5 quilogramas
     assert libras_quilos(0) == 0  # 0 libras = 0 quilogramas
-    assert libras_quilos(23.1495) == 10.5  # Valor não inteiro
+    assert libras_quilos(23.1495) == approx(10.5, rel=1e-4)  # Valor não inteiro
 
 def test_gramas_quilos():
     """Testa a conversão de gramas para quilogramas"""
     assert gramas_quilos(1000) == 1  # 1000 gramas = 1 quilograma
     assert gramas_quilos(5000) == 5  # 5000 gramas = 5 quilogramas
     assert gramas_quilos(0) == 0  # 0 gramas = 0 quilogramas
-    assert gramas_quilos(234) == 0.234  # Valor não inteiro
+    assert gramas_quilos(234) == approx(0.234, rel=1e-4)  # Valor não inteiro
 
 def test_quilos_gramas():
     """Testa a conversão de quilogramas para gramas"""
@@ -139,42 +140,42 @@ def test_minutos_para_horas():
 
 def test_mps_mph():
     """Testa a conversão de metros por segundo para milhas por hora"""
-    assert mps_mph(1) == 2.23694  # 1 metro por segundo = 2.23694 milhas por hora
-    assert mps_mph(5) == 11.1847  # 5 metros por segundo = 11.1847 milhas por hora
+    assert mps_mph(1) == approx(2.23694, rel=1e-4)  # 1 metro por segundo = 2.23694 milhas por hora
+    assert mps_mph(5) == approx(11.1847, rel=1e-4)  # 5 metros por segundo = 11.1847 milhas por hora
     assert mps_mph(0) == 0  # 0 metros por segundo = 0 milhas por hora
-    assert mps_mph(10.5) == 23.47437  # Valor não inteiro
+    assert mps_mph(10.5) == approx(23.47437, rel=1e-4)  # Valor não inteiro
 
 def test_mph_mps():
     """Testa a conversão de milhas por hora para metros por segundo"""
     assert mph_mps(2.23694) == 1  # 2.23694 milhas por hora = 1 metro por segundo
     assert mph_mps(11.1847) == 5  # 11.1847 milhas por hora = 5 metros por segundo
     assert mph_mps(0) == 0  # 0 milhas por hora = 0 metros por segundo
-    assert mph_mps(23.47437) == 10.5  # Valor não inteiro
+    assert mph_mps(23.47437) == approx(10.5, rel=1e-4)  # Valor não inteiro
 
 def test_kph_mps():
     """Testa a conversão de quilômetros por hora para metros por segundo"""
-    assert kph_mps(1) == 0.277778  # 1 quilômetro por hora = 0.277778 metros por segundo
-    assert kph_mps(5) == 1.38889  # 5 quilômetros por hora = 1.38889 metros por segundo
+    assert kph_mps(1) == approx(0.277778, rel=1e-4)  # 1 quilômetro por hora = 0.277778 metros por segundo
+    assert kph_mps(5) == approx(1.38889, rel=1e-4)  # 5 quilômetros por hora = 1.38889 metros por segundo
     assert kph_mps(0) == 0  # 0 quilômetros por hora = 0 metros por segundo
-    assert kph_mps(10.5) == 2.91667  # Valor não inteiro
+    assert kph_mps(10.5) == approx(2.91667, rel=1e-4)  # Valor não inteiro
 
 def test_mps_kph():
     """Testa a conversão de metros por segundo para quilômetros por hora"""
     assert mps_kph(1) == 3.6  # 1 metro por segundo = 3.6 quilômetros por hora
     assert mps_kph(5) == 18  # 5 metros por segundo = 18 quilômetros por hora
     assert mps_kph(0) == 0  # 0 metros por segundo = 0 quilômetros por hora
-    assert mps_kph(10.5) == 37.8  # Valor não inteiro
+    assert mps_kph(10.5) == approx(37.8, rel=1e-4)  # Valor não inteiro
 
 def test_kph_mph():
     """Testa a conversão de quilômetros por hora para milhas por hora"""
-    assert kph_mph(1) == 0.621371  # 1 quilômetro por hora = 0.621371 milhas por hora
-    assert kph_mph(5) == 3.106855  # 5 quilômetros por hora = 3.106855 milhas por hora
+    assert kph_mph(1) == approx(0.621371, rel=1e-4)  # 1 quilômetro por hora = 0.621371 milhas por hora
+    assert kph_mph(5) == approx(3.106855, rel=1e-4)  # 5 quilômetros por hora = 3.106855 milhas por hora
     assert kph_mph(0) == 0  # 0 quilômetros por hora = 0 milhas por hora
-    assert kph_mph(10.5) == 6.5244  # Valor não inteiro
+    assert kph_mph(10.5) == approx(6.5244, rel=1e-4)  # Valor não inteiro
 
 def test_mph_kph():
     """Testa a conversão de milhas por hora para quilômetros por hora"""
-    assert mph_kph(1) == 1.60934  # 1 milha por hora = 1.60934 quilômetros por hora
-    assert mph_kph(5) == 8.0467  # 5 milhas por hora = 8.0467 quilômetros por hora
+    assert mph_kph(1) == approx(1.60934, rel=1e-4)  # 1 milha por hora = 1.60934 quilômetros por hora
+    assert mph_kph(5) == approx(8.0467, rel=1e-4)  # 5 milhas por hora = 8.0467 quilômetros por hora
     assert mph_kph(0) == 0  # 0 milhas por hora = 0 quilômetros por hora
-    assert mph_kph(10.5) == 16.8989  # Valor não inteiro
+    assert mph_kph(10.5) == approx(16.8989, rel=1e-4)  # Valor não inteiro

--- a/tests/test_und_conv.py
+++ b/tests/test_und_conv.py
@@ -34,17 +34,17 @@ def test_kelvin_celsius():
 
 def test_fahrenheit_kelvin():
     """Testa a conversão de Fahrenheit para Kelvin"""
-    assert fahrenheit_kelvin(32) == approx(273.15, rel=1e-4)  # Ponto de congelamento da água
-    assert fahrenheit_kelvin(212) == approx(373.15, rel=1e-4)  # Ponto de ebulição da água
-    assert fahrenheit_kelvin(-40) == approx(233.15, rel=1e-4)  # Teste com valores negativos
-    assert fahrenheit_kelvin(98.6) == approx(310.15, rel=1e-4)  # Temperatura corporal média
+    assert fahrenheit_kelvin(32) == approx(273.15, rel=1e-3)  # Ponto de congelamento da água
+    assert fahrenheit_kelvin(212) == approx(373.15, rel=1e-3)  # Ponto de ebulição da água
+    assert fahrenheit_kelvin(-40) == approx(233.15, rel=1e-3)  # Teste com valores negativos
+    assert fahrenheit_kelvin(98.6) == approx(310.15, rel=1e-3)  # Temperatura corporal média
 
 def test_kelvin_fahrenheit():
     """Testa a conversão de Kelvin para Fahrenheit"""
-    assert kelvin_fahrenheit(273.15) == approx(32, rel=1e-4)  # Ponto de congelamento da água
-    assert kelvin_fahrenheit(373.15) == approx(212, rel=1e-4)  # Ponto de ebulição da água
-    assert kelvin_fahrenheit(0) == approx(-459.67, rel=1e-4)  # Zero absoluto
-    assert kelvin_fahrenheit(310.15) == approx(98.6, rel=1e-4)  # Temperatura corporal média
+    assert kelvin_fahrenheit(273.15) == approx(32, rel=1e-3)  # Ponto de congelamento da água
+    assert kelvin_fahrenheit(373.15) == approx(212, rel=1e-3)  # Ponto de ebulição da água
+    assert kelvin_fahrenheit(0) == approx(-459.67, rel=1e-3)  # Zero absoluto
+    assert kelvin_fahrenheit(310.15) == approx(98.6, rel=1e-3)  # Temperatura corporal média
 
 ####################### 2. Testes de Distância #######################
 
@@ -64,15 +64,15 @@ def test_quilometros_metros():
 
 def test_milhas_quilometros():
     """Testa a conversão de milhas para quilômetros"""
-    assert milhas_quilometros(1) == approx(1.60934, rel=1e-4)  # 1 milha = 1.60934 quilômetros
-    assert milhas_quilometros(5) == approx(8.0467, rel=1e-4)  # 5 milhas = 8.0467 quilômetros
+    assert milhas_quilometros(1) == approx(1.60934, rel=1e-3)  # 1 milha = 1.60934 quilômetros
+    assert milhas_quilometros(5) == approx(8.0467, rel=1e-3)  # 5 milhas = 8.0467 quilômetros
     assert milhas_quilometros(0) == 0  # 0 milhas = 0 quilômetros
-    assert milhas_quilometros(0.5) == approx(0.80467, rel=1e-4)  # Valor não inteiro
+    assert milhas_quilometros(0.5) == approx(0.80467, rel=1e-3)  # Valor não inteiro
 
 def test_quilometros_milhas():
     """Testa a conversão de quilômetros para milhas"""
-    assert quilometros_milhas(1) == approx(0.621371, rel=1e-4)  # 1 quilômetro = 0.621371 milhas
-    assert quilometros_milhas(5) == approx(3.106855, rel=1e-4)  # 5 quilômetros = 3.106855 milhas
+    assert quilometros_milhas(1) == approx(0.621371, rel=1e-3)  # 1 quilômetro = 0.621371 milhas
+    assert quilometros_milhas(5) == approx(3.106855, rel=1e-3)  # 5 quilômetros = 3.106855 milhas
     assert quilometros_milhas(0) == 0  # 0 quilômetros = 0 milhas
     assert quilometros_milhas(1.60934) == 1  # 1 milha = 1 quilômetro
 
@@ -80,24 +80,24 @@ def test_quilometros_milhas():
 
 def test_quilos_libras():
     """Testa a conversão de quilogramas para libras"""
-    assert quilos_libras(1) == approx(2.20462, rel=1e-4)  # 1 quilograma = 2.20462 libras
-    assert quilos_libras(5) == approx(11.0231, rel=1e-4)  # 5 quilogramas = 11.0231 libras
+    assert quilos_libras(1) == approx(2.20462, rel=1e-3)  # 1 quilograma = 2.20462 libras
+    assert quilos_libras(5) == approx(11.0231, rel=1e-3)  # 5 quilogramas = 11.0231 libras
     assert quilos_libras(0) == 0  # 0 quilograma = 0 libras
-    assert quilos_libras(10.5) == approx(23.1495, rel=1e-4)  # Valor não inteiro
+    assert quilos_libras(10.5) == approx(23.1495, rel=1e-3)  # Valor não inteiro
 
 def test_libras_quilos():
     """Testa a conversão de libras para quilogramas"""
-    assert libras_quilos(2.20462) == approx(1, rel=1e-4)  # 2.20462 libras = 1 quilograma
-    assert libras_quilos(11.0231) == approx(5, rel=1e-4)  # 11.0231 libras = 5 quilogramas
+    assert libras_quilos(2.20462) == approx(1, rel=1e-3)  # 2.20462 libras = 1 quilograma
+    assert libras_quilos(11.0231) == approx(5, rel=1e-3)  # 11.0231 libras = 5 quilogramas
     assert libras_quilos(0) == 0  # 0 libras = 0 quilogramas
-    assert libras_quilos(23.1495) == approx(10.5, rel=1e-4)  # Valor não inteiro
+    assert libras_quilos(23.1495) == approx(10.5, rel=1e-3)  # Valor não inteiro
 
 def test_gramas_quilos():
     """Testa a conversão de gramas para quilogramas"""
     assert gramas_quilos(1000) == 1  # 1000 gramas = 1 quilograma
     assert gramas_quilos(5000) == 5  # 5000 gramas = 5 quilogramas
     assert gramas_quilos(0) == 0  # 0 gramas = 0 quilogramas
-    assert gramas_quilos(234) == approx(0.234, rel=1e-4)  # Valor não inteiro
+    assert gramas_quilos(234) == approx(0.234, rel=1e-3)  # Valor não inteiro
 
 def test_quilos_gramas():
     """Testa a conversão de quilogramas para gramas"""
@@ -140,42 +140,41 @@ def test_minutos_para_horas():
 
 def test_mps_mph():
     """Testa a conversão de metros por segundo para milhas por hora"""
-    assert mps_mph(1) == approx(2.23694, rel=1e-4)  # 1 metro por segundo = 2.23694 milhas por hora
-    assert mps_mph(5) == approx(11.1847, rel=1e-4)  # 5 metros por segundo = 11.1847 milhas por hora
-    assert mps_mph(0) == 0  # 0 metros por segundo = 0 milhas por hora
-    assert mps_mph(10.5) == approx(23.47437, rel=1e-4)  # Valor não inteiro
+    assert mps_mph(1) == approx(2.23694, rel=1e-3)  # 1 m/s = 2.23694 mph
+    assert mps_mph(5) == approx(11.1847, rel=1e-3)  # 5 m/s = 11.1847 mph
+    assert mps_mph(0) == 0  # 0 m/s = 0 mph
+    assert mps_mph(10.5) == approx(23.47437, rel=1e-3)  # Valor não inteiro
 
 def test_mph_mps():
     """Testa a conversão de milhas por hora para metros por segundo"""
-    assert mph_mps(2.23694) == 1  # 2.23694 milhas por hora = 1 metro por segundo
-    assert mph_mps(11.1847) == 5  # 11.1847 milhas por hora = 5 metros por segundo
-    assert mph_mps(0) == 0  # 0 milhas por hora = 0 metros por segundo
-    assert mph_mps(23.47437) == approx(10.5, rel=1e-4)  # Valor não inteiro
+    assert mph_mps(2.23694) == approx(1, rel=1e-3)  # 2.23694 mph = 1 m/s
+    assert mph_mps(11.1847) == approx(5, rel=1e-3)  # 11.1847 mph = 5 m/s
+    assert mph_mps(0) == 0  # 0 mph = 0 m/s
 
 def test_kph_mps():
     """Testa a conversão de quilômetros por hora para metros por segundo"""
-    assert kph_mps(1) == approx(0.277778, rel=1e-4)  # 1 quilômetro por hora = 0.277778 metros por segundo
-    assert kph_mps(5) == approx(1.38889, rel=1e-4)  # 5 quilômetros por hora = 1.38889 metros por segundo
-    assert kph_mps(0) == 0  # 0 quilômetros por hora = 0 metros por segundo
-    assert kph_mps(10.5) == approx(2.91667, rel=1e-4)  # Valor não inteiro
+    assert kph_mps(1) == approx(0.277778, rel=1e-3)  # 1 km/h = 0.277778 m/s
+    assert kph_mps(5) == approx(1.38889, rel=1e-3)  # 5 km/h = 1.38889 m/s
+    assert kph_mps(0) == 0  # 0 km/h = 0 m/s
+    assert kph_mps(10.5) == approx(2.91667, rel=1e-3)  # Valor não inteiro
 
 def test_mps_kph():
     """Testa a conversão de metros por segundo para quilômetros por hora"""
-    assert mps_kph(1) == 3.6  # 1 metro por segundo = 3.6 quilômetros por hora
-    assert mps_kph(5) == 18  # 5 metros por segundo = 18 quilômetros por hora
-    assert mps_kph(0) == 0  # 0 metros por segundo = 0 quilômetros por hora
-    assert mps_kph(10.5) == approx(37.8, rel=1e-4)  # Valor não inteiro
+    assert mps_kph(1) == 3.6  # 1 m/s = 3.6 km/h
+    assert mps_kph(5) == 18  # 5 m/s = 18 km/h
+    assert mps_kph(0) == 0  # 0 m/s = 0 km/h
+    assert mps_kph(10.5) == approx(37.8, rel=1e-3)  # Valor não inteiro
 
 def test_kph_mph():
     """Testa a conversão de quilômetros por hora para milhas por hora"""
-    assert kph_mph(1) == approx(0.621371, rel=1e-4)  # 1 quilômetro por hora = 0.621371 milhas por hora
-    assert kph_mph(5) == approx(3.106855, rel=1e-4)  # 5 quilômetros por hora = 3.106855 milhas por hora
-    assert kph_mph(0) == 0  # 0 quilômetros por hora = 0 milhas por hora
-    assert kph_mph(10.5) == approx(6.5244, rel=1e-4)  # Valor não inteiro
+    assert kph_mph(1) == approx(0.621371, rel=1e-3)  # 1 km/h = 0.621371 mph
+    assert kph_mph(5) == approx(3.106855, rel=1e-3)  # 5 km/h = 3.106855 mph
+    assert kph_mph(0) == 0  # 0 km/h = 0 mph
+    assert kph_mph(10.5) == approx(6.5244, rel=1e-3)  # Valor não inteiro
 
 def test_mph_kph():
     """Testa a conversão de milhas por hora para quilômetros por hora"""
-    assert mph_kph(1) == approx(1.60934, rel=1e-4)  # 1 milha por hora = 1.60934 quilômetros por hora
-    assert mph_kph(5) == approx(8.0467, rel=1e-4)  # 5 milhas por hora = 8.0467 quilômetros por hora
-    assert mph_kph(0) == 0  # 0 milhas por hora = 0 quilômetros por hora
-    assert mph_kph(10.5) == approx(16.8989, rel=1e-4)  # Valor não inteiro
+    assert mph_kph(1) == approx(1.60934, rel=1e-3)  # 1 mph = 1.60934 km/h
+    assert mph_kph(5) == approx(8.0467, rel=1e-3)  # 5 mph = 8.0467 km/h
+    assert mph_kph(0) == 0  # 0 mph = 0 km/h
+    assert mph_kph(10.5) == approx(16.8989, rel=1e-3)  # Valor não inteiro

